docs/content/ingestion/native_tasks.md

```diff
@@ -181,7 +181,7 @@ The tuningConfig is optional and default parameters will be used if no tuningCon
 |reportParseExceptions|If true, exceptions encountered during parsing will be thrown and will halt ingestion; if false, unparseable rows and fields will be skipped.|false|no|
 |pushTimeout|Milliseconds to wait for pushing segments. It must be >= 0, where 0 means to wait forever.|0|no|
 |segmentWriteOutMediumFactory|Segment write-out medium to use when creating segments. See [SegmentWriteOutMediumFactory](#segmentWriteOutMediumFactory).|Not specified, the value from `druid.peon.defaultSegmentWriteOutMediumFactory.type` is used|no|
-|maxNumSubTasks|Maximum number of tasks which can be run at the same time.|Integer.MAX_VALUE|no|
+|maxNumSubTasks|Maximum number of tasks which can be run at the same time. The supervisor task would spawn sub tasks up to `maxNumSubTasks` regardless of the reamining task slots. If this value is set to too large, too many sub tasks can be created which might block other ingestion.|2|no|
```

Comment:
remaining*

Reply:
Thanks. I'll fix this and add a new section to emphasize the importance of setting this value properly.

Reply:
Added `Capacity Planning` section.

Reply:
Since the value is so critical and tough to set automatically, how about setting it to `1` by default and logging a warning in the task log when it's not raised higher? That way users are more likely to notice it if they didn't read the documentation.

Reply:
I thought about it, but set to 2 because I’m thinking to make the supervisor task does the ingestion job on its own instead of spawning sub tasks if this is 1. 

What do you think about more aggressive way like making this option mandatory and throwing error if it’s missing?

Reply:
> I thought about it, but set to 2 because I’m thinking to make the supervisor task do the ingestion job on its own instead of spawning sub tasks if this is 1.

If this is the case, I think 1 is an even better default.

> What do you think about more aggressive way like making this option mandatory and throwing error if it’s missing?

I think there should be some kind of default (we want to minimize the # of mandatory parameters) and at least "1" + a warning is something where it's fairly clear what's going on if you forgot to set it.

Reply:
Default of 1 with a warning message that tells the user to set "maxNumSubTasks" if they want parallelism sounds good to me.

Reply:
Changed default to 1 and added a warning. I also changed the behavior of default to the sequential mode to emphasize the importance.